src/test/java/org/folio/rest/impl/ReceivingHistoryTest.java

```diff
@@ -131,27 +117,21 @@ private void verifyViewCollectionAfter(URL endpoint) {
     List<ReceivingHistory> receivingHistories = receivingHistory.getReceivingHistory();
     for (int i = 0; i < receivingHistories.size(); i++) {
       if (receivingHistories.get(i).getId().equals(pieces[i].getId())) {
-        verifyFields(pieces[i], poLines[i], details[i], purchaseOrders[i], receivingHistories.get(i));
+        verifyFields(pieces[i], poLines[i], purchaseOrders[i], receivingHistories.get(i));
       } else {
-        verifyFields(pieces[i], poLines[i], details[i], purchaseOrders[i], receivingHistories.get(receivingHistories.size() - i -1));
+        verifyFields(pieces[i], poLines[i], purchaseOrders[i], receivingHistories.get(receivingHistories.size() - i -1));
       }
     }
   }
 
-  private void verifyFields(Piece piece, PoLine poLine, Details details, PurchaseOrder purchaseOrder, ReceivingHistory receivingHistory) {
-    assertEquals(piece.getCaption(), receivingHistory.getCaption());
-    assertEquals(piece.getComment(), receivingHistory.getComment());
-    assertEquals(piece.getItemId(), receivingHistory.getItemId());
-    assertEquals(piece.getLocationId(), receivingHistory.getLocationId());
-    assertEquals(piece.getSupplement(), receivingHistory.getSupplement());
-    assertEquals(poLine.getTitle(), receivingHistory.getTitle());
-    assertEquals(piece.getPoLineId(), receivingHistory.getPoLineId());
-    assertEquals(poLine.getPoLineNumber(), receivingHistory.getPoLineNumber());
-    assertEquals(details.getReceivingNote(), receivingHistory.getReceivingNote());
-    assertEquals(poLine.getPurchaseOrderId(), receivingHistory.getPurchaseOrderId());
-    assertEquals(purchaseOrder.getDateOrdered(), receivingHistory.getDateOrdered());
-    assertEquals(poLine.getOrderFormat().value(), receivingHistory.getPoLineOrderFormat().value());
-    assertEquals(poLine.getReceiptStatus().value(), receivingHistory.getPoLineReceiptStatus().value());
+  private void verifyFields(Piece piece, PoLine poLine, PurchaseOrder purchaseOrder, ReceivingHistory receivingHistory) {
+    assertEquals(receivingHistory.getCaption(), piece.getCaption());
+    assertEquals(receivingHistory.getComment(), piece.getComment());
+    assertEquals(receivingHistory.getItemId(), piece.getItemId());
+    assertEquals(receivingHistory.getSupplement(), piece.getSupplement());
+    assertEquals(receivingHistory.getTitle(), poLine.getTitle());
+    assertEquals(receivingHistory.getPoLineId(), piece.getPoLineId());
+    assertEquals(receivingHistory.getPoLineId(), poLine.getId());
```

Comment:
Please review and add missing assertions as needed

Reply:
I've added the missing assertions.  However, poLine.getDetails().getReceivingNote() is causing issues.

Reply:
Resolved issues with poLine.getDetails.getReceivingNote().